src/components/edit/Synonyms.jsx

```diff
@@ -1,14 +1,164 @@
 import React, { Component } from 'react';
-// import { test } from 'coda/services/agents';
+import PropTypes from 'prop-types';
+import Select from 'react-virtualized-select';
+import { ScientificName, Synonyms } from '../shared/partials.jsx';
+import { RadioGroup, TextInput, TextArea } from '../shared/FormInputs.jsx';
+import { getAgent, addOrUpdateSynonym } from 'coda/services/agents';
+import { arrayBufferToString } from 'coda/services/utils';
+import { BOOLEANS } from './constants';
+import autobind from 'react-autobind';
+
+let blankSynonym = {
+  genus: '',
+  species: '',
+  subSpecies: '',
+  authority: '',
+  isPrimary: false,
+  notes: ''
+};
 
 export default class EditSynonyms extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selected: undefined,
+      selectedAgent: undefined,
+      formattedSynonyms: [],
+      selectedSynonym: undefined,
+      newSynonym: false,
+      prevSynonym: undefined
+    };
+    autobind(this);
+  }
+
+  onAgentSelected(option) {
+    if (!option || !option.value) {
+      this.setState({
+        selected: undefined,
+        selectedAgent: undefined,
+        formattedSynonyms: [],
+        selectedSynonym: undefined,
+        newSynonym: false,
+        prevSynonym: undefined
+      });
+      return;
+    }
+    getAgent(option.value)
+      .then((agent) => {
+        let selectedSynonym = agent.synonyms.find((synonym) => {
+          return synonym.id === option.synId;
+        });
+        selectedSynonym = { ...selectedSynonym };
+        selectedSynonym.notes = arrayBufferToString(selectedSynonym.notes.data).replace(/ -/g, '\n-');
+        this.setState({ selectedAgent: agent, selectedSynonym: selectedSynonym, selected: option, newSynonym: false });
+      });
+  }
+
+  onSynonymSelected(option) {
+    if (!option || !option.value) {
+      this.setState({ selectedSynonym: null });
+      return;
+    }
+    this.setState({ selectedSynonym: option.value });
+  }
+
+  onSynonymChange(e) {
+    let updatedSynonym = { ...this.state.selectedSynonym, [e.target.name]: e.target.value };
+    this.setState({
```

Comment:
I think these `setState`s can all be one-liners, not sure whether the linter is telling you about them properly or not